rollout_service.rb

```diff
@@ -11,13 +11,18 @@ def authenticate!
         response = HTTParty.post('https://www.googleapis.com/oauth2/v3/tokeninfo',
                                  body: {id_token: params[:id_token]})
 
-        raise 'Bad response from server' if response.code != 200
+        raise "Bad response from server: #{response.inspect}" if response.code != 200
         response_body = JSON.parse(response.body)
 
         if $google_oauth_allowed_domain.present? && response_body['hd'] != $google_oauth_allowed_domain
           raise 'Unauthorized user, this domain is not allowed'
         end
 
+        allowed = (ENV['ALLOWED_EMAILS'] || '').split(',')
+        if !allowed.empty? && !allowed.include?(response_body['email'])
+          puts "#{response_body['email'].inspect} is not allowed (#{allowed.inspect})"
```

Comment:
Currently rollout_service/rollout_dashboard is allowing anyone with a Gmail account to make changes - which is not what I'd want an admin tool. This change fixes it, it makes it check the user account against allowed emails list set in ALLOWED_EMAILS env var.